src/test/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/service/RecallServiceTest.kt

```diff
@@ -308,6 +340,88 @@ class RecallServiceTest {
+    every { nicStillUalRecentlyUpdatedRecall.status() } returns Status.AWAITING_RETURN_TO_CUSTODY
+    every { nicStillUalRecentlyUpdatedRecall.nomsNumber } returns nicStillUalRecentlyUpdatedNoms
+    every { nicStillUalRecentlyUpdatedRecall.lastUpdatedDateTime } returns OffsetDateTime.now(fixedClock).minusMinutes(10)
+
+    val nicStillUalPreviouslyUpdatedNoms = NomsNumber("ZYX9876W")
+    val nicStillUalPreviouslyUpdatedRecall = mockk<Recall>()
+    every { nicStillUalPreviouslyUpdatedRecall.status() } returns Status.AWAITING_RETURN_TO_CUSTODY
+    every { nicStillUalPreviouslyUpdatedRecall.nomsNumber } returns nicStillUalPreviouslyUpdatedNoms
+    every { nicStillUalPreviouslyUpdatedRecall.lastUpdatedDateTime } returns OffsetDateTime.now(fixedClock).minusMinutes(returnToCustodyUpdateThresholdMinutes - 10)
+
+    val nicStillUalRecentlyUpdatedPrisoner = mockk<Prisoner>()
+    every { nicStillUalRecentlyUpdatedPrisoner.status } returns "NOT ACTIVE"
+
+    val nicRtcRecall = existingRecall.copy(
+      lastUpdatedDateTime = OffsetDateTime.now(fixedClock).minusMinutes(returnToCustodyUpdateThresholdMinutes + 10),
+      assessedByUserId = UUID.randomUUID(),
+      inCustodyAtAssessment = false,
+      warrantReferenceNumber = WarrantReferenceNumber("ABC12345/AB")
+    )
+    val rtcNoms = nicRtcRecall.nomsNumber
+    val nicRtcPrisoner = mockk<Prisoner>()
+    every { nicRtcPrisoner.status } returns "ACTIVE IN"
+
+    val recallList = listOf(nicRtcRecall, nicStillUalRecentlyUpdatedRecall, nicStillUalPreviouslyUpdatedRecall, inCustodyRecall)
+
+    val returnedToCustodyDateTime = OffsetDateTime.now().minusHours(15)
+    val movementDate = returnedToCustodyDateTime.toLocalDate()
+    val movementTime = returnedToCustodyDateTime.toLocalTime()
+    val dossierTargetDate = OffsetDateTime.now(fixedClock).plusDays(1).toLocalDate()
+    val expectedRecall = nicRtcRecall.copy(
+      returnedToCustody = ReturnedToCustodyRecord(
+        returnedToCustodyDateTime,
+        OffsetDateTime.now(fixedClock),
+        OffsetDateTime.now(fixedClock),
+        RecallService.SYSTEM_USER_ID
+      ),
+      dossierTargetDate = dossierTargetDate
+    )
+
+    every { bankHolidayService.nextWorkingDate(OffsetDateTime.now(fixedClock).toLocalDate()) } returns dossierTargetDate
+    every { recallRepository.findAll() } returns recallList
+    every { prisonerOffenderSearchClient.prisonerByNomsNumber(nicStillUalPreviouslyUpdatedNoms) } returns Mono.just(nicStillUalRecentlyUpdatedPrisoner)
+    every { prisonerOffenderSearchClient.prisonerByNomsNumber(rtcNoms) } returns Mono.just(nicRtcPrisoner)
+
+    every { prisonApiClient.latestInboundMovements(setOf(rtcNoms)) } returns listOf(Movement(rtcNoms.value, movementDate, movementTime))
+    every { recallRepository.save(expectedRecall, RecallService.SYSTEM_USER_ID) } returns expectedRecall
+
+    underTest.updateCustodyStatus(currentUserId)
+
+    verify { recallRepository.save(expectedRecall, RecallService.SYSTEM_USER_ID) }
```

Comment:
Wondering if any further `verify` calls might be justified?  E.g. to `bankHolidayService`?  
🤔  Would this test fail if zero calls were made matching the `every { bankHolidayService.nextWorkingDate ...` etc. calls?

Reply:
It's annoying you cant force the every to also verify